storey/targets.py

```diff
@@ -625,16 +627,23 @@ class StreamTarget(Flow, _Writer):
         inferred from data and used in place of explicit columns list if none was provided, or appended to the provided list. If header is
         True and columns is not provided, infer_columns_from_data=True is implied. Optional. Default to False if columns is provided,
         True otherwise.
-    :param shard_count: If stream doesn't exist, it will be created with this number of shards. Defaults to 1.
+    :param shards: If stream doesn't exist, it will be created with this number of shards. Defaults to 1.
     :param retention_period_hours: If stream doesn't exist, it will be created with this retention time in hours. Defaults to 24.
     :param storage_options: Extra options that make sense for a particular storage connection, e.g. host, port, username, password, etc.,
         if using a URL that will be parsed by fsspec, e.g., starting “s3://”, “gcs://”. Optional
     :type storage_options: dict
     """
 
-    def __init__(self, storage: Driver, stream_path: str, sharding_func: Optional[Callable[[Event], int]] = None, batch_size: int = 8,
-                 columns: Optional[List[str]] = None, infer_columns_from_data: Optional[bool] = None,
-                 shard_count: int = 1, retention_period_hours: int = 24, full_event: Optional[bool] = None, **kwargs):
+    def __init__(self,
+                 storage: Driver, stream_path: str,
+                 sharding_func: Union[None, int, str, Callable[[Event], Any]] = None,
```

Comment:
instead of Any, can be Union[int,str]

Reply:
It doesn't need to be an int or a string though.